feature/intern/src/test/java/com/terning/intern/InternViewModelTest.kt

```diff
@@ -0,0 +1,63 @@
+package com.terning.intern
+
+import com.terning.core.designsystem.state.UiState
+import com.terning.domain.intern.entity.InternInfo
+import com.terning.domain.intern.repository.InternRepository
+import com.terning.feature.intern.InternViewModel
+import com.watcha.testing.util.MainDispatcherRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.mock
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class InternViewModelTest {
+
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+
+    private lateinit var viewModel: InternViewModel
+    private val internRepository: InternRepository = mock()
+
+    @Before
+    fun setUp() {
+        viewModel = InternViewModel(internRepository)
+    }
+
+    @Test
+    fun `getInternInfo 성공 시 UiState_Success 설정`() = runTest {
+        val mockInternInfo = InternInfo(
+            dDay = "10",
+            title = "테스트 제목",
+            deadline = "테스트 내용",
+            workingPeriod = "3",
+            startYearMonth = "2023.01",
+            scrapCount = 10,
+            viewCount = 10,
+            company = "테스트 제목",
+            companyCategory = "테스트 내용",
+            companyImage = "테스트 내용",
+            qualification = "테스트 내용",
+            jobType = "테스트 내용",
+            detail = "테스트 내용",
+            url = "테스트 내용",
+            isScrapped = true
+        )
+
+        doReturn(Result.success(mockInternInfo))
+            .`when`(internRepository).getInternInfo(1)
+
+        viewModel.getInternInfo(1)
+        advanceUntilIdle()
+
+        val uiState = viewModel.internUiState.first()
+        assert(uiState.loadState is UiState.Success)
+        assert((uiState.loadState as UiState.Success).data == mockInternInfo)
```

Comment:
```suggestion
        assertTrue(uiState.loadState is UiState.Success)
        assertEquals((uiState.loadState as UiState.Success).data, mockInternInfo)
```
이렇게 작성도 가능하니 한번 써 보는 것도 좋을 것 같아요😊

제가 테스트코드 공부할 때는 기본 assert()는 잘 안 썼던 것 같아서 왜 그런지 찾아보니 가독성 측면이 큰 것 같더라구요..!! 테스트 코드는 유지보수를 위한 코드인 만큼 가독성이 중요해서 다양한 assertion 함수를 사용하게 되었다고 합니다ㅎㅎ

Reply:
우와 확실히 코드 가독성이 훨씬 좋아지는 것 같아요!!!!! 감사합니당